moped-editor/src/views/projects/projectView/ProjectComponents/BasemapSpeedDial.js

```diff
@@ -0,0 +1,163 @@
+import {
+  MAPBOX_PADDING_PIXELS,
+  MAPBOX_CONTROL_BUTTON_WIDTH,
+  COLORS,
+} from "./mapStyleSettings";
+
+const useStyles = makeStyles((theme) => ({
+  speedDialAction: {
+    width: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    height: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    backgroundSize: "100% 100%",
+    borderRadius: 4,
+    background: COLORS.mutedGray,
+  },
+  speedDialStreets: {
+    color: "black",
+    backgroundImage: `url(${process.env.PUBLIC_URL}/static/images/mapStreets.jpg)`,
+  },
+  speedDialAerial: {
+    color: "white",
+    backgroundImage: `url(${process.env.PUBLIC_URL}/static/images/mapAerial.jpg)`,
+  },
+  fabLabel: ({ isSpeedDialOpen }) => ({
+    ...(isSpeedDialOpen && { backgroundColor: "rgba(0,0,0,.25)" }),
+    height: "100%",
+  }),
+  mapStyleToggleLabel: {
+    fontSize: ".8rem",
+    fontWeight: "bold",
+  },
+  mapStyleToggleLabelIcon: {
+    position: "relative",
+    top: 3,
+    marginRight: "4px",
+  },
+  mapStyleActionLabel: {
+    position: "absolute",
+    bottom: 0,
+  },
+  speedDial: {
+    position: "absolute",
+    height: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    right: `${MAPBOX_PADDING_PIXELS}px`,
+    // Mapbox basemap has copyright info below the speedial while NearMap tiles do not
+    bottom: ({ basemapKey }) =>
+      basemapKey === "aerial"
+        ? `${MAPBOX_PADDING_PIXELS}px`
+        : `${MAPBOX_PADDING_PIXELS * 3}px`,
+    // Mapbox copyright info collapses to a taller info icon at 990px and below
+    [theme.breakpoints.down(991)]: {
+      bottom: ({ basemapKey }) =>
+        basemapKey === "aerial"
+          ? `${MAPBOX_PADDING_PIXELS}px`
+          : `${MAPBOX_PADDING_PIXELS * 4}px`,
+    },
+  },
+}));
+
+/**
+ * Generates the basemap selector using the SpeedDial component
+ * @param {function} setBasemapKey - The function we will call with to change the basemap key value
+ * @param {string} basemapKey - The name of the current basemap key
+ * @return {JSX.Element}
+ */
+const BasemapSpeedDial = ({ setBasemapKey, basemapKey }) => {
+  const [isSpeedDialOpen, setIsSpeedDialOpen] = useState(false);
+  const classes = useStyles({ basemapKey, isSpeedDialOpen });
+
+  /**
+   * Changes the current basemap and closes the speed dial menu
+   * @param basemapName
+   */
+  const onBasemapSelect = (basemapKey) => {
+    setIsSpeedDialOpen(false);
+    setBasemapKey(basemapKey);
+  };
+
+  /**
+   * Opens the speed dial menu
+   */
+  const onOpen = () => {
+    setIsSpeedDialOpen(true);
+  };
+
+  const onClose = () => setIsSpeedDialOpen(false);
+
+  return (
+    <SpeedDial
+      className={classes.speedDial}
+      ariaLabel="Basemap Select"
+      hidden={false}
+      icon={
+        <Typography>
+          <Icon
+            className={clsx(
+              classes.mapStyleToggleLabel,
+              classes.mapStyleToggleLabelIcon
+            )}
+          >
+            layers
+          </Icon>
+          <span className={classes.mapStyleToggleLabel}>Map</span>
```

Comment:
Would you consider striking this label bit, the `Map` bit? If the stacked-layers icon were able to grow to be bigger in the speed dial, I think they would better convey what you adjust with it.

Reply:
I like this a lot. Going to push an update since I agree that the **Map** text doesn't add a lot of meaning here.